src/core/Akka.Tests/Actor/FailingActorSpec.cs

```diff
@@ -0,0 +1,490 @@
+            {
+                child = Context.ActorOf(new RandomPool(1).WithSupervisorStrategy(strategy).Props(Props.Create<FailingActor>()));
+
+                Receive<string>(m => child.Tell(m), m => true);
+
+                resetEvt.Set();
+            }
+
+            private void CreatePooledChildConsistentHashing()
+            {
+                child = Context.ActorOf(new ConsistentHashingPool(1).WithHashMapping(o => o.GetHashCode()).WithSupervisorStrategy(strategy).Props(Props.Create<FailingActor>()));
+
+                Receive<string>(m => child.Tell(m), m => true);
+
+                resetEvt.Set();
+            }
+
+            private void CreatePooledChildTailChopping()
+            {
+                child = Context.ActorOf(new TailChoppingPool(1, TimeSpan.FromSeconds(10), TimeSpan.FromMilliseconds(20)).WithSupervisorStrategy(strategy).Props(Props.Create<FailingActor>()));
+
+                Receive<string>(m => child.Tell(m), m => true);
+
+                resetEvt.Set();
+            }
+
+            private void CreatePooledChildScatterGatherFirstCompleted()
+            {
+                child = Context.ActorOf(new ScatterGatherFirstCompletedPool(1).WithSupervisorStrategy(strategy).Props(Props.Create<FailingActor>()));
+
+                Receive<string>(m => child.Tell(m), m => true);
+
+                resetEvt.Set();
+            }
+
+            private void CreatePooledChildSmallestMailbox()
+            {
+                child = Context.ActorOf(new SmallestMailboxPool(1).WithSupervisorStrategy(strategy).Props(Props.Create<FailingActor>()));
+
+                Receive<string>(m => child.Tell(m), m => true);
+
+                resetEvt.Set();
+            }
+
+            private void CreateGroupedChildBroadcast()
+            {
+                IActorRef realChild = Context.ActorOf(Props.Create<FailingActor>(), "failing");
+
+                child = Context.ActorOf(Props.Empty.WithRouter(new BroadcastGroup(new string[] { realChild.Path.ToString() })).WithSupervisorStrategy(strategy));
+
+                Receive<string>(m => child.Tell(m), m => true);
+
+                resetEvt.Set();
+            }
+
+            private void CreateGroupedChildRoundrobin()
+            {
+                IActorRef realChild = Context.ActorOf(Props.Create<FailingActor>(), "failing");
+
+                child = Context.ActorOf(Props.Empty.WithRouter(new BroadcastGroup(new string[] { realChild.Path.ToString() })).WithSupervisorStrategy(strategy));
+
+                Receive<string>(m => child.Tell(m), m => true);
+
+                resetEvt.Set();
+            }
+
+            private void CreateGroupedChildRandom()
+            {
+                IActorRef realChild = Context.ActorOf(Props.Create<FailingActor>(), "failing");
+
+                child = Context.ActorOf(Props.Empty.WithRouter(new BroadcastGroup(new string[] { realChild.Path.ToString() })).WithSupervisorStrategy(strategy));
```

Comment:
Sorry if I got it wrong, but I think that ParentActor is already doing what you do point in its SupervisorStrategy() at this [method](https://github.com/akkadotnet/akka.net/pull/2433/files/abdc679a0f6e4b41f75363a85bff9a5ad305fab7#diff-1c1211befa61f9dbf239d8c8b525ba74R217). I see that the  WithSupervisorStrategy method does not have any effect on group routers, but ParentActor handles it in this case. Am I missing something?

I am making some changes in order to check whether Decider is hit or not for all of the test cases, will commit it along with your requested changes - based on your response to this comment.